src/GraphQL.Tests/Types/AutoRegisteringInterfaceGraphTypeTests.cs

```diff
@@ -0,0 +1,809 @@
+    }
+
+    private interface NullSourceTest
+    {
+        public static bool Example1 { get; set; } = true;
+        public static string Example2() => "test";
+        public static int Example3 = 3;
+    }
+#endif
+
+    [Fact]
+    public void TestExceptionBubbling()
+    {
+        Should.Throw<Exception>(() => new AutoRegisteringInterfaceGraphType<TestExceptionBubblingInterface>()).Message.ShouldBe("Test");
+    }
+
+    [Fact]
+    public void TestBasicClassNoExtraFields()
+    {
+        var graphType = new AutoRegisteringInterfaceGraphType<TestBasicInterface>();
+        graphType.Fields.Find("Id").ShouldNotBeNull();
+        graphType.Fields.Find("Name").ShouldNotBeNull();
+        graphType.Fields.Count.ShouldBe(2);
+    }
+
+    [Fact]
+    public async Task CustomHardcodedArgumentAttributesWork()
+    {
+        var graphType = new AutoRegisteringInterfaceGraphType<CustomHardcodedArgumentAttributeTestInterface>();
+        var fieldType = graphType.Fields.Find(nameof(CustomHardcodedArgumentAttributeTestInterface.FieldWithHardcodedValue))!;
+        var resolver = fieldType.Resolver!;
+        (await resolver.ResolveAsync(new ResolveFieldContext
+        {
+            Source = new CustomHardcodedArgumentAttributeTestClass(),
+        }).ConfigureAwait(false)).ShouldBe("85");
+    }
+
+    [Fact]
+    public void TestInheritedFields()
+    {
+        var graphType = new AutoRegisteringInterfaceGraphType<InheritanceTests>();
+        graphType.Fields.Find("Field1").ShouldNotBeNull();
+        graphType.Fields.Find("Field2").ShouldNotBeNull();
+        graphType.Fields.Find("Field3").ShouldNotBeNull();
+        graphType.Fields.Find("Field4").ShouldNotBeNull();
+        graphType.Fields.Find("Field5").ShouldNotBeNull();
+        graphType.Fields.Count.ShouldBe(5);
+    }
+
+    [Theory]
+    [InlineData("{find(type:CAT){id name}}", @"{""data"":{""find"":{""id"":""10"",""name"":""Fluffy""}}}")]
+    [InlineData("{find(type:DOG){id name}}", @"{""data"":{""find"":{""id"":""20"",""name"":""Shadow""}}}")]
+    [InlineData("{find(type:CAT){id name ... on Cat { lives } ... on Dog { isLarge }}}", @"{""data"":{""find"":{""id"":""10"",""name"":""Fluffy"",""lives"":9}}}")]
+    [InlineData("{find(type:DOG){id name ... on Cat { lives } ... on Dog { isLarge }}}", @"{""data"":{""find"":{""id"":""20"",""name"":""Shadow"",""isLarge"":true}}}")]
+    [InlineData("{cat{id name}}", @"{""data"":{""cat"":{""id"":""10"",""name"":""Fluffy""}}}")]
+    [InlineData("{dog{id name}}", @"{""data"":{""dog"":{""id"":""20"",""name"":""Shadow""}}}")]
+    [InlineData("{cat{...frag}} fragment frag on IAnimal { id name }", @"{""data"":{""cat"":{""id"":""10"",""name"":""Fluffy""}}}")]
+    [InlineData("{dog{...frag}} fragment frag on IAnimal { id name }", @"{""data"":{""dog"":{""id"":""20"",""name"":""Shadow""}}}")]
+    public async Task EndToEndTest(string query, string expected)
```

Comment:
These tests demonstrate scenario described in the migration notes and tests typical behavior when writing a type-first schema.